flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = { nixpkgs, rust-overlay, ... }:
+    let
+      system = "x86_64-linux";
+
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [ rust-overlay.overlays.default ];
+      };
+
+      rust-toolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+    in
+    {
```

Comment:
We can add the following for enabling people to use flake to package flake version of rio using as following:
```nix
 packages = eachSystem (system: {
      default = self.packages.${system}.rio;
      inherit
        (pkgsFor.${system})
         rio
         # other dependencies
        ;
    });
```

Reply:
hm... I don't understand what this exactly does or why we should do this. Will other users be able to run include a rio package from this repo (as a "nightly" version of rio) or what is the benefit of this?

Reply:
Yes, one wpuld be able to lock on to a specific rev from the git when using flakes and pass rio as an input. Expected result would be that of a nightly release.